draco/analysis/sidereal.py

```diff
@@ -682,7 +682,7 @@ def process(self, sstream: containers.SiderealStream) -> containers.SiderealStre
         # Helpful for creating generic configs
         try:
             era = sstream.effective_ra[:].local_array
-        except KeyError:
+        except AttributeError:
```

Comment:
I think the `effective_ra` property will indeed return a `KeyError` if the dataset is missing:
```
    @property
    def effective_ra(self):
        """Get the effective_ra dataset if it exists, None otherwise."""
        if "effective_ra" in self.datasets:
            return self.datasets["effective_ra"]

        raise KeyError("Dataset 'effective_ra' not initialised.")
```

Reply:
Good catch. I'm not sure why I made that change.